src/solver/rosenbrock.cu

```diff
@@ -247,7 +288,7 @@ namespace micm
 
       if (number_of_elements != errors_param.number_of_elements_)
       {
-        throw std::runtime_error("devstruct.errors_input_ and errors_param have different sizes.");
+        throw std::system_error(make_error_code(MicmCudaRosenbrockSolverErrc::ParameterMismatch), "");
```

Comment:
Might be worth having a message why it's mismatched for easier debugging?

Reply:
I ended up switching this to use the new `INTERNAL_ERROR()` because it looked to me more like a check for a bug in our code. But, if that's not correct, I can change it back to throw one of our user errors